### DISCUSSION TITLE
Gradio progressbar affected generation speed

### DISCUSSION THREAD

**Discussion by Irmagon** [lllyasviel/Fooocus#2326] [Q&A] [answered]
Setup: 
Fooocus version: 2.1.865
Total VRAM 12287 MB, total RAM 65484 MB
Set vram state to: NORMAL_VRAM
Always offload VRAM
Device: cuda:0 NVIDIA GeForce RTX 3060 : native
Browser: Chrome with hardware acceleraton enabe

Step 1: maximized browser window - vcard loading ~30% GPU-3D
![image](https://github.com/lllyasviel/Fooocus/assets/36661892/dcde5033-68b0-45b9-bfbc-3c30df60b6cb)
![image](https://github.com/lllyasviel/Fooocus/assets/36661892/078d9da1-160b-4614-8fe1-7a033688a2ff)

Step 2: press Generation and minimise browser window - improve generation speed (2.3 sec faster) and no loading GPU-3D
![image](https://github.com/lllyasviel/Fooocus/assets/36661892/8346db92-b6a8-4990-abe9-5eea7ccbe5dc)
![image](https://github.com/lllyasviel/Fooocus/assets/36661892/30f9ed5c-14b9-4473-bfb8-5ccfd027b34c)

I check other browser - Opera, Cent browser, Firefox...
And it's the same again (except for Firefox, which has a GPU-3D load ower 50%), if hardware accelerated enable.

In base Gradio parameters are **--no-progressbar-hiding** (do not hide progressbar in gradio UI (we hide it because it slows down ML if you have hardware accleration in browser) (**default: False**))

**How to disable gradio progressbar?**
In Fooocus, progressbar always on (or I didn't find it in the settings?)

**Comment by mashb1t** [comment]
I can't reproduce what you're experiencing on my setup.
![image](https://github.com/lllyasviel/Fooocus/assets/9307310/b7a6d0c7-31ab-42b1-b948-44af3296b93f)

Fooocus doesn't use the regular progress bar, but a synthetic one with gradio yielding.
You can disable gradio updates to progress bar as well as intermediate images completely by commenting the code in https://github.com/lllyasviel/Fooocus/blob/main/webui.py#L57-L61
Please check if applying https://github.com/lllyasviel/Fooocus/pull/1013 works for you (didn't for me), which drastically reduces the traffic / updates on the gradio intermediate output.
Hope this helps in narrowing down the issue.

**Comment by Irmagon** [comment]
> Fooocus doesn't use the regular progress bar, but a synthetic one with gradio yielding. You can disable gradio updates to progress bar as well as intermediate images completely by commenting the code in https://github.com/lllyasviel/Fooocus/blob/main/webui.py#L57-L61 Please check if applying #1013 works for you (didn't for me), which drastically reduces the traffic / updates on the gradio intermediate output. Hope this helps in narrowing down the issue.

I try (((
No effects, GPU-3D still loadings. Disable hardware acceleration in browser settings also not effected.
I try full disable output progressbar in webui.py and modules/html - no effects

**Reply by mashb1t** [reply]
But then this has nothing to do with the progress bar at all. Are you using the same config / doing the same generation with same params in each browser? Can you use Chrome in incognito mode to ensure extensions aren't an issue?
In general, high GPU 3D usage indicates full GPU power used for generating the image, but i am a bit clueless (as you are) why this is not sn issue when minimizing the browser.

**Reply by Irmagon** [reply]
Manuel, i'm sorry, it looks like the progress bar really isn't involved.

I reinstall clear Chrome, and started checking the progress bar and preview window.
I change, for checking, https://github.com/lllyasviel/Fooocus/blob/main/webui.py#L59 to 
`gr.update(visible=True, value=image) if image is not None else gr.update(visible=False), \`
to make it easier to check the display of the preview window.
When I activate "Disable Preview", the preview window is hidden during generadion and the load on GPU-3D = 0.
I checked before and with commented out lines L57-L61, the preview window displayed without an image, but the load GPU-3D was still there.
And now I think the problem is the fkn animated border of the Gradio preview window.
It does make such a load when rendering glowing border in the browser.

Maybe it's really my local problem...

**Reply by Irmagon** [reply]
And performance slow down with spinner around progress bar
![image](https://github.com/lllyasviel/Fooocus/assets/36661892/bcc754b7-c363-4526-8847-9ac45a61ae78)

After remove `<div class="loader"></div>` in html.py
![image](https://github.com/lllyasviel/Fooocus/assets/36661892/b0d04087-498f-403a-b40a-92b293312d78)

This is not my local problem, its checked on 6 different machines my frends

**Reply by Irmagon** [reply, marked as answer]
How to improve performance?

1. Disable spinner in progress-bar: delete `<div class="loader"></div>`  from \modules\html.py
2. Disable gloving border in gradio:
- search **index-29fa5a20.css** in (for portable installation or search this in you pithon installed folders)
  \python_embeded\Lib\site-packages\gradio\templates\cdn\assets\
  \python_embeded\Lib\site-packages\gradio\templates\frontend\assets\
- decompact it (i.e. on https://www.cleancss.com/css-beautify/) 
- change CSS to left panel view and save
![image](https://github.com/lllyasviel/Fooocus/assets/36661892/0b2b48c3-71c3-4d6c-a353-47a822807cd8)
- or replace with this file (download and change extension to CSS)
  [index-29fa5a20.txt](https://github.com/lllyasviel/Fooocus/files/14970481/index-29fa5a20.txt)

**Comment by Spide85** [comment]
I have a similar issue, for me it's the orange glowing border. If I remove it, I get a 10-20% increased it/s rate. Tested on Firefox and Chrome, similar results.
RTX3080OC + AMD 7800X3D

**Reply by Irmagon** [reply]
How u remove it?
I change index-29fa5a20.css in gradio folders.

**Reply by Spide85** [reply]
I just made it temporary with the dev-tools from the browser. Currently I just switch tabs while generating. Totally stupid, maybe we need an official fix? I have this on Linux and Windows with FF an chrome.
I have no idea about Gradio, so writing a fix/PR is not in my hands.
